### PR TITLE
feat(pet-126): hot-apply config edits to the live session

### PR DESCRIPTION
**Pull request by ziomancer**
## Summary
- Adds `Pipeline.reconfigure(new_config)` — one propagation primitive that atomically swaps `_config` and pushes the new config into every subcomponent that cached it (FrequencyTracker / AuditEmitter / AlertManager / MinimalScanner decode), without resetting accumulated session state.
- Extends the PET-111 cross-process idiom: a new `console/_reload.py` TTL + `(mtime_ns, size)`-gated reader and gateway wiring so the Hermes gateway picks up a `config.yaml` change on a running session (pipeline + guard + lineage + egress), fail-safe to last-good.
- Routes `ConsoleHandlers.update_config` through `reconfigure` so the standalone console path is fixed by the same primitive (no more bare `self.pipeline._config = validated`).

Closes the bug where an operator loosened a frequency/escalation/audit/alert knob mid-session and saw nothing change: those fields silently did not take effect until a new session.

## Layered defense / atomicity
- **In-process**: `reconfigure` merge-preserves the restart-required `session_secret` (warn-on-differing-non-None, never a silent half-apply), applies the only fail-prone step (the `frequency_weights` re-parse) **first** so a malformed map aborts before any state is touched, then the infallible steps, then commits the swap + re-resolves the profile with constructor precedence.
- **Cross-object (gateway)**: `_apply_reconfigure` is two-phase — `guard.validate_config` (D8 overlap + weight trial, no mutation) before any commit, then `pipeline.reconfigure` + `guard.apply_config` + `lineage.apply_config` + egress recompute, as one loop-serialized unit. `session_secret` is held immutable at both the tracker and guard level (FREQ-03).
- **D-WIN**: a wiped/empty `petasos:` section is a no-op (keep last-good), never an all-defaults reset. Peek + commit-on-success means a failed apply is retried, never silently pinning a stale config.

## Files changed

| File | Change |
|------|--------|
| `petasos/pipeline.py` | Adds `reconfigure`; stores `_profile_arg` for re-resolution |
| `petasos/session/frequency.py` | Adds `apply_config` (staged weight parse, commit under `_lock`, immutable secret, tombstone-cap trim) |
| `petasos/session/guard.py` | Adds `validate_config` + `apply_config`; `SpawnBudget.set_window` |
| `petasos/session/lineage.py` | Adds `apply_config` (rebind bounds, preserve edges) |
| `petasos/session/audit.py`, `alerting.py` | Adds `apply_config` (rebind `_config`; alerting rebuilds the PII ring buffer on capacity change keeping newest) |
| `petasos/scanners/minimal.py` | Adds `set_decode_encoded_payloads` (in-place, preserves identity/tunables) |
| `petasos/console/_reload.py` | New: cross-process change reader (peek + commit-on-success, same-tick guard, D-WIN) |
| `docs/deployment/reference_plugin/__init__.py` | `_build_config_from_section`, `_apply_reconfigure`, `_maybe_reconfigure`, rate-limited attribution/failure logs |
| `petasos/console/server.py` | `update_config` routes through `reconfigure` |
| `tests/test_pipeline_reconfigure.py`, `test_guard_reconfigure.py`, `test_console_reload.py`, `test_console_server.py` | New regression tests |
| `tests/test_reference_plugin_egress.py`, `test_subagent_plugin_wiring.py` | Neutralize the new reconfigure check in pre-existing `_pre_tool_call` paths |

## Test plan
- [x] `C:\python310\python.exe -m pytest -p no:cacheprovider -k "not test_default_ignorable_rejected"` — 1705 passed, 41 skipped, 1 deselected, 1 xfailed (full output: docs/specs/TODO/PET-126.test-output.txt)
- [x] `ruff check .` — clean
- [x] `ruff format --check .` — clean
- [x] `mypy --strict .` — clean
- [x] `test_default_ignorable_rejected` deselected (pre-existing local 3.10 / Unicode-13 failure, unrelated)

## Security note (PET-126 Decision 8)
This widens the PET-111 live re-read from one bit to the whole hot-swappable config. It adds **no** new guard enforcement and must not regress the PET-125 config-out-of-reach boundary (doc-led; guard path-blocklisting was rejected). Each applied cross-process reconfigure emits a rate-limited attribution WARNING. Do not present the broadened re-read as fully shipped without PET-125's boundary in view.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Live configuration hot-reload: detects changes and reconfigures the running pipeline, guards, and related components without restart.
  * Runtime reconfiguration now updates frequency tuning, guard validation/delegate rules, audit verbosity, alert thresholds/ring buffer capacity, and scanner decode behavior.
  * Environment overlays support for secrets/hash settings during configuration builds.
* **Bug Fixes**
  * Reload/apply failures keep the last known-good configuration and avoid blocking tool calls; reload failure warnings are rate-limited.
* **Tests**
  * Added coverage for reload change detection (including edge cases), atomic multi-phase reconfiguration, secret preservation, and rate-limited failure logging.

### CODE REVIEW

File: petasos/session/alerting.py [hunk 1]
```diff
@@ -54,6 +54,25 @@ def __init__(
...
+        ``alert_ring_buffer_capacity`` changed.
+
+        On a capacity *shrink*, ``deque(old, maxlen=new)`` keeps the most-recent
+        ``new`` items (oldest dropped): ring-buffer recency semantics. All
+        counters, cooldowns, per-rule/per-session deques, ``_ring_buffers``, and
+        ``_cross_session_tracker`` are preserved. The capacity is pre-validated as
+        a positive int in ``__post_init__``, so the rebuild cannot raise. Existing
+        per-rule ``_ring_buffers`` entries keep their construction-time ``maxlen``
+        (harmless: ``__post_init__`` bounds ``alert_rapid_fire_count <= capacity``);
+        newly-created entries pick up the new capacity.
```
[the remark this answers is not in the document]
Comment (ziomancer) [reply petasos/session/alerting.py:68]: __post_init__ bounds rapid_fire_count <= capacity so an over-sized buffer can't produce false positives, and the buffers self-correct as they age out via _prune_stale. Rebuilding all per-rule buffers on every capacity change adds complexity for zero correctness gain. (The _pii_ring_buffer is rebuilt on capacity change, which is the one that matters for the PII-volume threshold.)
